Documentation/general/cli-ux-guidelines.md

```diff
@@ -0,0 +1,155 @@
+.NET Core Command-Line Tools UX Guidelines
+-------------------------------------------
+
+This document will cover the CLI UX guidelines that anyone adding or suggesting
+a new command should take into consideration. It does not deal with
+implementation details. The main goal of the document is to provide guidelines
+for new authors that wish to make pull requests and add new functionalities and
+commands to the .NET Core command line tools (CLI).
+
+## Naming the commands
+
+In the .NET Core CLI, commands should be **verbs**. This rule was adopted
+because most of the commands do *something*.
+
+Sub-commands are supported in the CLI, and they are usually nouns. A good
+example of this is the “dotnet add reference” command. If there is a need to add
+a subcommand, that subcommand should usually specialize what the parent command
+does.
+
+## CRUD commands
+If a new CRUD command is to be added, its naming should be determined with the
+following logic:
+
+-   Does the command work on the data in the project (either properties or
+    items)? If yes, then it should be added as a noun to the “dotnet

```

Comment:
> then it should be added as a noun to the “dotnet add/list/remove/update”,

That feels awkward to me. After reading this a few times, I finally get it. You need to describe that there are established top-level (verb) commands and this is the right place for adding sub-commands that fall into the category you are talking about. As written, this is much too subtle, IMO.